### PR TITLE
feat(salsah1): Add case for boolean type in extended search

### LINKED ISSUE #1017: no search for boolean values
oct-29-2018 15-01-39

when selecting a boolean in the advanced search window, the list for possible comparison operator is empty

### PR DESCRIPTION
**Pull request by loicjaouen**
fixes #1017

### CODE REVIEW

**Comment by loicjaouen** [comment]
we can now search for resources for which a boolean value exist

**Comment by loicjaouen** [comment]
bool_search

**Comment by loicjaouen** [comment]
it works now :)

**Comment by tobiasschweizer** [comment]
Do you think you could add a test for this?

File: webapi/scripts/graphdb-knora-test-data.expect [hunk 1]
```diff
@@ -24,14 +24,10 @@ send "load ../_test_data/all_data/system-data.ttl into http://www.knora.org/data
 expect $prompt
 send "load ../_test_data/ontologies/incunabula-onto.ttl into http://www.knora.org/ontology/0803/incunabula .\r"
 expect $prompt
-send "load ../_test_data/all_data/incunabula-data.ttl into http://www.knora.org/data/0803/incunabula .\r"
```
Comment (subotic) [inline webapi/scripts/graphdb-knora-test-data.expect]: Any reason why you would need to remove the test data?

Comment (loicjaouen) [reply webapi/scripts/graphdb-knora-test-data.expect]: damned... no, that's local config committed inadvertently, fixing this

**Comment by loicjaouen** [comment]
@tobiasschweizer adding tests; is it worth the effort for a piece of software that is doomed to vanish in rather sooner than later?

**Comment by loicjaouen** [comment]
@tobiasschweizer apparently it is not obvious to run tests... what do I do wrong this time?

First step, running Sipi... the config is not on sipi anymore, updating the doc to point to where it is installed.

**Comment by loicjaouen** [comment]
right now I can't test it because docker hub is down: 
```
docker: Error response from daemon: Get https://registry-1.docker.io/v2/dhlabbasel/sipi/manifests/develop: received unexpected HTTP status: 503 Service Unavailable.
```

**Comment by tobiasschweizer** [comment]
it is documented here: https://docs.knora.org/paradox/06-salsah/index.html

you have to run Sipi with`sipi.knora-docker-test-config.lua`
the sipi files have been moved here: https://github.com/dhlab-basel/Knora/tree/develop/sipi

@subotic may perhaps help with the dockerhub issue

**Comment by loicjaouen** [comment]
@tobiasschweizer  thanks for the hint, but that's what I described in the previous commit f205a21

for docker hub... it is docker hub's issue, nothing to do but to wait for them to get back on their feet

**Comment by tobiasschweizer** [comment]
@loicjaouen sorry, missed that one!

I remember that I tried to run the tests some weeks ago and there was a problem. I fixed them some months, it was just a small issue with the order of the props

let me know if I can help

**Comment by loicjaouen** [comment]
let's bootstrap the triplestore:
- going for a dockerized version of graphdb-free:
```
docker run --rm -p 7200:7200 dhlabbasel/graphdb-free
```
- bootstrapping the test data with the script:
```
./graphdb-free-docker-init-knora-test.sh
```
- running knora:
```
sbt
> reStart -r
[...]
webapi [2018-10-30 17:02:28,174] INFO  - KnoraService(akka://webapi) - Please initialize repository. Will exit now.
[...]
```

looking up the config, in the section `triplestore` on finds:
```
    triplestore {
        dbtype = "graphdb-se"
```

but there is no `graphdb-se` config, rather:

```
        graphdb {
            port = 7200
            repository-name = "knora-test"
            username = "admin"
            password = "root"
        }
```

switching `dbtype` to:  `dbtype = "graphdb-se"`, we get:
```
sbt:webapi> webapi [2018-10-30 17:11:32,456] ERROR - OneForOneStrategy - Embedded triplestore type graphdb not supported
```

and switching the both to `graphdb-se` we get:

```
webapi[ERROR] Caused by: com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'app.triplestore.graphdb'
```

**Comment by loicjaouen** [comment]
finally running the tests:
```
[info] Run completed in 9 minutes, 10 seconds.
[info] Total number of tests run: 28
[info] Suites: completed 3, aborted 0
[info] Tests: succeeded 9, failed 19, canceled 0, ignored 1, pending 0
[info] *** 19 TESTS FAILED ***
[error] Failed tests:
[error] 	org.knora.salsah.browser.ResourceCreationSpec
[error] 	org.knora.salsah.browser.SearchAndEditSpec
```

Apparently knora can connect at first, but then fails:

```
webapi ================================================================
webapi Knora API Server started at http://0.0.0.0:3333
webapi ----------------------------------------------------------------
webapi WARNING: Resetting Triplestore Content over HTTP is turned ON.
webapi ----------------------------------------------------------------
webapi DB-Name: knora-test
webapi DB-Type: graphdb-free
webapi DB Server: localhost, DB Port: 7200
webapi ================================================================
webapi
webapi [2018-10-30 17:28:20,124] ERROR - Dispatcher - Reset: Failed to execute insert into triplestore
```

**Comment by tobiasschweizer** [comment]
@loicjaouen Do you want me to try on my machine?

**Comment by loicjaouen** [comment]
thanks for the proposal @tobiasschweizer , to avoid the many variables that can differ, could you try this:

- run the stack as described on https://docs.knora.org/paradox/05-internals/development/docker-compose.html (i configured it with a graphdb license)
- run the set-up script `./graphdb-se-docker-init-knora-test.sh`
- restart the stack
- run the salsah1.5 tests

different runs give these results:
> Tests: succeeded 10, failed 18, canceled 0, ignored 1, pending 0
> Tests: succeeded 9, failed 19, canceled 0, ignored 1, pending 0

and also could you remind me how to run only one of those tests?

**Comment by loicjaouen** [comment]
this way knora is not started with `-r` and can't load test data :(

**Comment by tobiasschweizer** [comment]
in sbt: testOnly *ResourceCreationSpec

**Comment by tobiasschweizer** [comment]
for ResourceCreationSpec I get one failing test, I guess it's the order of the props since everything is indexed by numbers and not Iris

**Comment by tobiasschweizer** [comment]
ResourceCreationSpec: there seems to be a problem with geoname id. Was it added to the ontology recently?

**Comment by loicjaouen** [comment]
it is there since the code was moved to github 3 years ago ;)

**Comment by tobiasschweizer** [comment]
mostly the GUI tests breaks because new properties are added to the test ontologies and then displayed in the GUI

**Comment by tobiasschweizer** [comment]
@loicjaouen If you need my help with this, I can dedicate some time to it next week.

**Comment by loicjaouen** [comment]
thanks @tobiasschweizer , I also had to switch to other issues for the time being

**Comment by tobiasschweizer** [comment]
@loicjaouen you mean you waited until we fixed it ;-)

just joking ... :-)

**Pull request closed without merging**